src/Microsoft.OpenApi/Validations/ValidationContext.cs

```diff
@@ -3,76 +3,39 @@
 
 using System;
 using System.Collections.Generic;
+using System.Linq;
+using Microsoft.OpenApi.Services;
 using Microsoft.OpenApi.Validations.Rules;
 
 namespace Microsoft.OpenApi.Validations
 {
     /// <summary>
-    /// The validation context.
+    /// Constrained interface used to provide context to rule implementation
     /// </summary>
-    public class ValidationContext
+    public interface IValidationContext 
     {
-        private readonly IList<ValidationError> _errors = new List<ValidationError>();
 
         /// <summary>
-        /// Initializes the <see cref="ValidationContext"/> class.
+        /// Register an error with the validation context.
         /// </summary>
-        /// <param name="ruleSet"></param>
-        public ValidationContext(ValidationRuleSet ruleSet)
-        {
-            RuleSet = ruleSet ?? throw Error.ArgumentNull(nameof(ruleSet));
-        }
+        /// <param name="error">Error to register.</param>
+        void AddError(ValidationError error);
 
         /// <summary>
-        /// Gets the rule set.
+        /// Allow Rule to indicate validation error occured a deeper context level.  
         /// </summary>
-        public ValidationRuleSet RuleSet { get; }
+        /// <param name="segment"></param>
+        void Enter(string segment);
 
         /// <summary>
-        /// Gets the validation errors.
+        /// Exit from path context elevel.  Enter and Exit calls should be matched.
         /// </summary>
-        public IEnumerable<ValidationError> Errors
-        {
-            get
-            {
-                return _errors;
-            }
-        }
+        void Exit();
```

Comment:
Why are these needed in the context?

Reply:
Because often the rules validate additional depth.  Rules add segments for properties in objects being validated.  Also, for custom validation rules may validate several levels of object depth.

---
In reply to: [162479370](https://github.com/Microsoft/OpenAPI.NET/pull/176#discussion_r162479370) [](ancestors = 162479370)